### DISCUSSION TITLE
Stop push workflow to run on pull request

### DISCUSSION THREAD

**Discussion by Icsyx** [actions/checkout#1588] [Q&A]
Let's say that we have the following scenario:

There is a Repository that have the following structure regarding the branches:
- develop
- feature/1
- feature/2
- ....

Inside this repository are 2 workflows: `push.yaml` and `pull_request.yaml`
First workflow triggers every time a push occurs since it have this trigger: 
`on:
    push:`
Second workflows triggers every time a Pull Request is opened or changed since it have this trigger: 
`on:
    pull_request:
        types: [opened, reopened, edited, review_requested, synchronize]`

When I open a Pull Request from a feature branch to develop both workflows triggers. By pushing a commit again triggers both workflows. The thing is that I don't want the `push` workflow to trigger here. I want it to trigger on every other push except here.
Is there any solution in preventing the `push.yaml` workflow to run when a commit is pushed for the Pull Request?

The only solution that I could find is [this](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-a-fallback-value). The problem with this is that it cancels the push workflow and mark it as failure. By doing that I am not able to merge the changes since not all checks are successful.

**Comment by ShreeChaturvedi** [comment]
Limit `push.yaml` to the branches you care about (`develop` / `main`). `push` and `pull_request` are separate events, so a feature-branch push still starts the push workflow unless you filter branches.

```yaml
# push.yaml
on:
  push:
    branches:
      - develop
      - main
```

Keep PR validation on the PR workflow:

```yaml
# pull_request.yaml
on:
  pull_request:
    types: [opened, reopened, edited, review_requested, synchronize]
    branches:
      - develop
```

Feature-branch pushes then only run the PR workflow. Direct pushes to `develop`/`main` run the push workflow.

Pitfall: `if: github.event_name != 'pull_request'` does nothing in a push-only file (`event_name` is always `push`). Cancel/fail hacks also break required checks. Prefer branch filters so `push.yaml` never starts for feature branches. If that workflow is a required check, stop requiring it on PRs or use the PR workflow instead.
